File: app/src/main/java/pl/laskowski/marcin/creating/ClassicSudokuGenerator.java
package pl.laskowski.marcin.creating;

import pl.laskowski.marcin.model.Sudoku;
import pl.laskowski.marcin.playground.TestSet;
import pl.laskowski.marcin.type.ClassicSquare;

/**
 * Created by Marcin Laskowski.
 */

public class ClassicSudokuGenerator extends SudokuGenerator {

    private final SudokuShuffler shuffler;
    private final Sudoku template;

    public ClassicSudokuGenerator(SudokuRater.Difficulty difficulty, Float percentFilled) {
        super(new ClassicSquare(9), difficulty, percentFilled);
        this.shuffler = new SudokuShuffler();
        this.template = TestSet.TEST.getData().get(0);
    }

    @Override
    protected Sudoku getFilledSudoku() {
        return shuffler.shuffleFull(template.copy(), new ClassicSquare(9));
    }

}

File: app/src/main/java/pl/laskowski/marcin/creating/ClassicSudokuGenerator.kt
package pl.laskowski.marcin.creating

import pl.laskowski.marcin.creating.SudokuRater.Difficulty
import pl.laskowski.marcin.model.Sudoku
import pl.laskowski.marcin.model.type.ClassicSquare9x9

class ClassicSudokuGenerator(difficulty: Difficulty?, percentFilled: Float?) :
    SudokuGenerator(ClassicSquare9x9, difficulty, percentFilled) {

    private val shuffler: SudokuShuffler = SudokuShuffler()

    override val filledSudoku: Sudoku
        get() = shuffler.shuffleFull(template.copy(), ClassicSquare9x9)

    companion object {
        private val template = Sudoku(ClassicSquare9x9, listOf(
            1,2,3, 4,5,6, 7,8,9,
            4,5,6, 7,8,9, 1,2,3,
            7,8,9, 1,2,3, 4,5,6,

            9,1,2, 3,4,5, 6,7,8,
            3,4,5, 6,7,8, 9,1,2,
            6,7,8, 9,1,2, 3,4,5,

            8,9,1, 2,3,4, 5,6,7,
            2,3,4, 5,6,7, 8,9,1,
            5,6,7, 8,9,1, 2,3,4
        ))
    }
}